public/app/features/live/live.ts

```diff
@@ -141,20 +161,163 @@ export class CentrifugeSrv implements GrafanaLiveSrv {
   //----------------------------------------------------------
 
   /**
-   * Is the server currently connected
+   * Listen for changes to the connection state
    */
-  isConnected() {
-    return this.centrifuge.isConnected();
+  getConnectionState() {
+    return this.connectionState.asObservable();
   }
 
   /**
-   * Listen for changes to the connection state
+   * Get a channel.  If the scope, namespace, or path is invalid, a shutdown
+   * channel will be returned with an error state indicated in its status.
+   *
+   * This is a singleton instance that stays active until explicitly shutdown.
+   * Multiple requests for this channel will return the same object until
+   * the channel is shutdown
    */
-  getConnectionState() {
-    return this.connectionState.asObservable();
+  async getChannelInfo(addr: LiveChannelAddress): Promise<LiveChannelConfig> {
+    const scope = this.scopes[addr.scope];
+    if (!scope) {
+      return Promise.reject('invalid scope');
+    }
+
+    const support = await scope.getChannelSupport(addr.namespace);
+    if (!support) {
+      return Promise.reject(addr.namespace + ' does not support streaming');
+    }
+    return support.getChannelConfig(addr.path)!;
+  }
+
+  /**
+   * Watch for messages in a channel
+   */
+  getStream<T>(address: LiveChannelAddress): Observable<LiveChannelEvent<T>> {
+    return this.getChannel<T>(address).getStream();
+  }
+
+  /**
+   * Connect to a channel and return results as DataFrames
+   */
```

Comment:
NOTE:  this is just moved from an external function to one that is directly on the liveSrv